src/Agent/NewRelic/Agent/Core/Configuration/ServerConfiguration.cs

```diff
@@ -332,4 +342,15 @@ public void OnError(StreamingContext context, ErrorContext errorContext)
             Log.ErrorFormat("Json serializer context path: {0}. Error message: {1}", errorContext.Path, errorContext.Error.Message);
         }
     }
+
+    /// <summary>
+    /// Provides options controlling the creation of <see cref="ServerConfiguration"/> instances.
+    /// </summary>
+    public class ServerConfigurationCreationOptions
```

Comment:
I wonder if it is necessary to create a class for this. I could be as simple as a bool.

Reply:
Debated that myself - this could be used to support more options in the future without needing to tack on more flag params and is more explicit in the code since you need to spell out the param name where as `call(param1, true)` lacks a certain amount of context unless you use named params for the bool. However, if this will be the only param for a long while an extra bool would suffice.